### PR TITLE
AC: per data instance metrics

### PR DESCRIPTION
**Pull request by eaidova**
- [x]  reuse metric `update()` method to get following results where it is possible
- [x] design metric instance result for callback
- [x] move outputs callback from launcher to model evaluator (for access to the metrics results)
- [x] adopt pipeline evaluator
- [x] tricks for full dataset detection metrics
- [x] tests

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins please retry a build

**Comment by eaidova** [comment]
Jenkins please retry a build

File: tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py [hunk 1]
```diff
@@ -166,19 +181,24 @@ def process_dataset(
...
             if self.metric_executor:
-                self.metric_executor.update_metrics_on_batch(annotations, predictions)
+                metrics_result = self.metric_executor.update_metrics_on_batch(batch_input_ids, annotations, predictions)
+                if self.metric_executor.need_store_predictions:
+                    self._annotations.extend(annotations)
+                    self._predictions.extend(predictions)
 
-            self._annotations.extend(annotations)
-            self._predictions.extend(predictions)
+            output_callback(batch_raw_predictions, metrics_result=metrics_result)
```
Comment (LeonidBeynenson) [inline tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py]: 2 @eaidova:  sorry, the case `output_callback is None` is possible here? probably,  the check `if output_callback` is required?

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py]: @LeonidBeynenson, thank you, fixed

File: tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py [hunk 2]
```diff
@@ -116,19 +115,34 @@ def _create_subset(subset, num_images):
...
-                        self.metric_executor.update_metrics_on_batch(annotations, predictions)
+                        metrics_result = self.metric_executor.update_metrics_on_batch(
+                            batch_input_ids, annotations, predictions
+                        )
                         if self.metric_executor.need_store_predictions:
                             self._annotations.extend(annotations)
                             self._predictions.extend(predictions)
 
+                    if output_callback:
+                        output_callback(batch_raw_predictions, metrics_result=metrics_result)
```
Comment (LeonidBeynenson) [inline tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py]: 2 @eaidova:  Ekaterina, could you also pass to `output_callback` a list that will allow to match for each element in the batch
* "element identifier" -- i.e. the string that identifies the element in the dataset (most probably they are in `batch_identifiers`, aren't they?)
* "element index" -- i.e. the number of the element in the dataset

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py]: done,
added `element_identifiers` argument for `batch_identifiers` and `dataset_indices` for dataset elements numeration.

**Comment by eaidova** [comment]
Jenkins please retry a build

**Pull request merged**